### PR TITLE
Fixes in ImageFileTest and RasterYuv420Semiplanar

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Fixes https://github.com/cogneco/ooc-kean/issues/450
Corrected conversion function in RasterYuvSemiplanar, if source height was even, this function could write outside the result buffer.
Simplified conversion function in RasterBgra.
Fixes some minor memory leaks related to not `free`d closures.
Cleaned up `ImageFileTest`, added `expect` calls to test if the image file was created.
Added `free` calls on images in `ImageFileTest`, as on Linux there was a crash in `RasterYuvSemiplanar` destructor when image height was not even (this thing crashed sooner on Windows).

### CODE REVIEW

File: source/draw/RasterBgra.ooc [hunk 1]
```diff
@@ -37,22 +37,23 @@ RasterBgra: class extends RasterPacked {
 		result
 	}
 	apply: func ~bgr (action: Func(ColorBgr)) {
-		end := this buffer pointer as Long + this buffer size
-		rowLength := this size width * this bytesPerPixel
-		for (row: Long in this buffer pointer as Long .. end) {
-			rowEnd := row + rowLength
-			for (source: Long in row .. rowEnd) {
-				action((source as ColorBgr*)@)
-				source += 3
```
Comment (sebastianbaginski) [inline source/draw/RasterBgra.ooc]: I believe this was not correct in case of bgra images, as each pixel has `4` components, not `3`. It uses `this bytesPerPixel` now.

Comment (davidhesselbom) [inline source/draw/RasterBgra.ooc]: I think the idea was that `source` was incremented automatically by the `for` loop, hence `+3`.

File: source/draw/RasterBgra.ooc [hunk 2]
```diff
@@ -37,22 +37,23 @@ RasterBgra: class extends RasterPacked {
...
+		for (row in 0 .. this size height) {
+			source := this buffer pointer + row * this stride
```
Comment (davidhesselbom) [inline source/draw/RasterBgra.ooc]: `row * this stride` should be the same on each row, but whatever.

File: source/draw/RasterBgra.ooc [hunk 3]
```diff
@@ -37,22 +37,23 @@ RasterBgra: class extends RasterPacked {
...
+				action(pixelPointer@)
 			}
-			row += this stride - 1
 		}
 	}
 	apply: func ~yuv (action: Func(ColorYuv)) {
-		this apply(ColorConvert fromBgr(action))
+		convert := ColorConvert fromBgr(action)
+		this apply(convert)
+		(convert as Closure) dispose()
```
Comment (davidhesselbom) [inline source/draw/RasterBgra.ooc]: Hm, why was this change necessary?

Comment (sebastianbaginski) [inline source/draw/RasterBgra.ooc]: *valgrind* reported a leak here. Very small, but still.

File: source/draw/RasterBgra.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/draw/RasterBgra.ooc]: Oh yeah, you're right :+1: 
Too many casts anyway, imho.

File: source/draw/RasterBgra.ooc (hunk 3 above)
Comment (davidhesselbom) [inline source/draw/RasterBgra.ooc]: Weird. Did the old version a closure implicitly, I wonder?

File: source/draw/RasterBgra.ooc (hunk 1 above)
Comment (davidhesselbom) [inline source/draw/RasterBgra.ooc]: Agreed, this looks much better now.
